CMakeLists.txt

```diff
@@ -0,0 +1,470 @@
+    "${PROJECT_SOURCE_DIR}/util/crc32c.h"
+    "${PROJECT_SOURCE_DIR}/util/env_posix.cc"
+    "${PROJECT_SOURCE_DIR}/util/env.cc"
+    "${PROJECT_SOURCE_DIR}/util/filter_policy.cc"
+    "${PROJECT_SOURCE_DIR}/util/hash.cc"
+    "${PROJECT_SOURCE_DIR}/util/hash.h"
+    "${PROJECT_SOURCE_DIR}/util/logging.cc"
+    "${PROJECT_SOURCE_DIR}/util/logging.h"
+    "${PROJECT_SOURCE_DIR}/util/mutexlock.h"
+    "${PROJECT_SOURCE_DIR}/util/options.cc"
+    "${PROJECT_SOURCE_DIR}/util/posix_logger.h"
+    "${PROJECT_SOURCE_DIR}/util/random.h"
+    "${PROJECT_SOURCE_DIR}/util/status.cc"
+
+  # Only CMake 3.3+ supports PUBLIC sources in targets exported by "install".
+  $<$<VERSION_GREATER:CMAKE_VERSION,3.2>:PUBLIC>
+    "include/leveldb/c.h"
+    "include/leveldb/cache.h"
+    "include/leveldb/comparator.h"
+    "include/leveldb/db.h"
+    "include/leveldb/dumpfile.h"
+    "include/leveldb/env.h"
+    "include/leveldb/filter_policy.h"
+    "include/leveldb/iterator.h"
+    "include/leveldb/options.h"
+    "include/leveldb/slice.h"
+    "include/leveldb/status.h"
+    "include/leveldb/table_builder.h"
+    "include/leveldb/table.h"
+    "include/leveldb/write_batch.h"
+)
+target_include_directories(leveldb
+  PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+)
+target_compile_definitions(leveldb
+  PRIVATE
+    LEVELDB_PLATFORM_POSIX
+)
+
+# TODO(pwnall): This is only needed for port_posix.
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+target_link_libraries(leveldb Threads::Threads)
+
+if (HAVE_CRC32C)
+  target_link_libraries(leveldb crc32c)
+endif (HAVE_CRC32C)
+if (HAVE_SNAPPY)
+  target_link_libraries(leveldb snappy)
+endif (HAVE_SNAPPY)
+
+add_library(leveldb_memenv "")
+target_sources(leveldb_memenv
+  PRIVATE
+    "${PROJECT_SOURCE_DIR}/helpers/memenv/memenv.cc"
+
+  # Only CMake 3.3+ supports PUBLIC sources in targets exported by "install".
+  $<$<VERSION_GREATER:CMAKE_VERSION,3.2>:PUBLIC>
+    "${PROJECT_SOURCE_DIR}/helpers/memenv/memenv.h"
+)
+target_compile_definitions(leveldb_memenv
+  PRIVATE
+    LEVELDB_PLATFORM_POSIX
+)
+target_link_libraries(leveldb_memenv leveldb)
+
+add_executable(leveldbutil
+  "${PROJECT_SOURCE_DIR}/db/leveldbutil.cc"
+)
+target_link_libraries(leveldbutil leveldb)
+
+if(LEVELDB_BUILD_TESTS)
+  enable_testing()
+
+  add_library(leveldb_test_base "")
+  target_sources(leveldb_test_base
+    PRIVATE
+      "${PROJECT_SOURCE_DIR}/util/testharness.cc"
+      "${PROJECT_SOURCE_DIR}/util/testutil.cc"
+    PUBLIC
+      "${PROJECT_BINARY_DIR}/include/port/port_config.h"
+      "${PROJECT_SOURCE_DIR}/util/testharness.h"
+      "${PROJECT_SOURCE_DIR}/util/testutil.h"
+  )
+  target_compile_definitions(leveldb_test_base
+    PUBLIC
+      LEVELDB_PLATFORM_POSIX
+  )
+  target_link_libraries(leveldb_test_base leveldb)
+
+  add_executable(autocompact_test
+    "${PROJECT_SOURCE_DIR}/db/autocompact_test.cc"
```

Comment:
Relative path will work fine here, you can use `db/autocompact_test.cc`.

Reply:
Agreed